### ISSUE TITLE
Get random records from findMany and also with pagination

### ISSUE THREAD

**Issue by hyochan** [prisma/prisma#5894]
## Problem

Comming from https://github.com/prisma/prisma/discussions/5886.

As discussed in [Prisma 1 forum](https://v1.prisma.io/forum/t/query-for-random-record/975/9), looks like there is no way to find random records with `findMany`. Indeed, this is such a challenge to use with `pagination`.

## Suggested solution

Providing a `random` function from `prisma` would be an easy solution for users therefore we can expect results with `orderBy: random()`. Although this results in a bad performance in pure `sql`, I think `Prisma` can do better internally without just exposing the random inside the `sql`.

## Alternatives

If the below pagination query and raw query could aggregate, it would scope down many limitations.

```
prisma.item.findMany({
...relayToPrismaPagination({
  after,
  before,
  first,
  last,
}),
  where: {
    ...filters,
  },
  orderBy: {id: 'desc'},
});
```

```
select * from "Item" limit 20
offset floor(random() * (select count(*) from "Item"));
```

**Comment by perroudsky** [comment]
@hyochan 

My temporary solution is to add a 'randomizer' field to your model. 
When you create your item, you store a random string into that 'randomizer'.
Then in your query you order it by adding orderBy: {randomizer: 'desc'} for exemple.

I've used short-uuid to generate my random string.

Not perfect of course, mainly because even if randomised, the items are always returned in the same order (but I guess its necessary for pagination so...).

Hope this help :)

**Comment by naothomachida** [comment]
Looking for it

**Comment by DraftProducts** [comment]
Is this feature planned?
There is still no way today to retrieve a row random in the database with Prisma.

**Comment by ryancwalsh** [comment]
I've been checking back on [this issue](https://github.com/prisma/prisma1/issues/114) and [these docs](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sorting) since June 2020 or earlier, and I see that others have wanted this functionality since Feb 2017 or earlier.

@schickling I'm curious whether you think anyone will work on this soon?

Also, I propose an optional `seed: number` argument (in addition to the `first: number` argument [proposed here](https://github.com/prisma/prisma1/issues/114#issuecomment-285677243)).

Sometimes it's helpful to let the client generate and store a seed (such as from a random number generator) that it can then pass to this backend function, which will then return the results sorted in a way that will always be the same (given that the full array of results hasn't changed in the meantime and the same seed was provided).

See the following example of some shuffling code I've used ([inspired](https://stackoverflow.com/a/53758827/470749) by Mike Bostock's implementation of the Fisher–Yates algorithm):

```TS
/**
 *
 * @param {array} array
 * @param {number} seed
 * @returns {array}
 */
export function shuffle(array: any, seed: number) {
  // https://stackoverflow.com/a/53758827/

  const result = [...array];
  let a = result.length;
  let b;
  let elementToMove;
  let currentSeed = seed;

  // While there remain elements to shuffle…
  while (a) {
    // Pick a remaining element…
    a -= 1;
    b = Math.floor(random(currentSeed) * a);

    // And swap it with the current element.
    elementToMove = result[a];
    result[a] = result[b];
    result[b] = elementToMove;
    currentSeed += 1;
  }

  return result;
}

```

**Comment by deemeetree** [comment]
Yes, we absolutely need this feature. Please, consider adding it into the next upgrades.

**Comment by ansh** [comment]
Would be super interested in this feature.

**Comment by kaloyanBozhkov** [comment]
I am also interested in this.

**Comment by Zenoo** [comment]
Yep, still needed on my end

**Comment by hjKangIB** [comment]
I'll wait this function. really need this.

**Comment by kblizeck** [comment]
Upvote here too - would be really helpful

**Comment by najibghadri** [comment]
UP

**Comment by iamthe-Wraith** [comment]
upvote here as well

**Comment by EliasTouil** [comment]
I will find a workaround but I am also very interested in this feature.

**Comment by bhupesh98** [comment]
I required this feature too, please tag me whenever there's update to this issue. Absence of advanced SQL functions is making me dismiss the usage of prisma in further projects.

**Comment by GauravBR** [comment]
Is there any way to get only one random value after adding initial filter like where:{ }?

**Comment by toknT** [comment]
Yeah , it's really needed . 100% of my project need it even it's not necessary in first release. Using prisma from 2022.

**Comment by rahulvijay5** [comment]
We can just use alternate menthods, but we can have proper dedicated method for it, Kindly consider this in next update please.

**Comment by hakkikonu** [comment]
+1

**Comment by theromanticsoul** [comment]
+1
